collie/interactions/datasets.py

```diff
@@ -333,45 +342,70 @@ def __init__(self,
+                if self._checked_num_negative_item_samples_is_valid is False:
+                    self._checked_num_negative_item_samples_is_valid = (
+                        self._check_num_negative_samples_is_valid()
+                    )
+            elif self.negative_sample_type == 'user':
+                if self._checked_num_negative_user_samples_is_valid is False:
+                    self._checked_num_negative_user_samples_is_valid = (
+                        self._check_num_negative_samples_is_valid()
+                    )
+
+        self.positive_sets = {}
+        if self.max_number_of_samples_to_consider > 0:
+            print(f'Generating positive {self.negative_sample_type}s set...')
+            self._generate_positive_set()
+
+    def _check_num_negative_samples_is_valid(self) -> bool:
+        """Check if the number of negative samples is valid."""
+        if self.negative_sample_type == 'item':
             counter = collections.Counter(self.mat.row)
-            max_number_of_items_interacted_with = counter.most_common(1)[0][1]
-            print('Maximum number of items a user has interacted with: {}'.format(
-                max_number_of_items_interacted_with
-            ))
+            num_ids = self.num_items
+            opposite_type = 'user'
+        elif self.negative_sample_type == 'user':
+            counter = collections.Counter(self.mat.col)
+            num_ids = self.num_users
+            opposite_type = 'item'
+
+        max_number_interacted_with = counter.most_common(1)[0][1]
+        print(
+            f'Maximum number of {self.negative_sample_type}s a {opposite_type} has interacted with:'
+            f' {max_number_interacted_with}'
+        )
 
-            del counter
+        del counter
 
-            is_valid = (
-                self.num_negative_samples
-                < (self.num_items - max_number_of_items_interacted_with)
-            )
-            assert is_valid, '``num_negative_samples`` must be less than {}!'.format(
-                (self.num_items - max_number_of_items_interacted_with)
-            )
+        is_valid = (
+            self.num_negative_samples
+            < (num_ids - max_number_interacted_with)
+        )
+        assert is_valid, '``num_negative_samples`` must be less than {}!'.format(
+            (num_ids - max_number_interacted_with)
+        )
 
-        self.positive_items = {}
-        if self.max_number_of_samples_to_consider > 0:
-            print('Generating positive items set...')
-            self._generate_positive_item_set()
+        return True
 
-    def _generate_positive_item_set(self) -> None:
-        """Build positive item dictionary lookup for exact negative sampling."""
-        self.positive_items = set(zip(self.mat.row, self.mat.col))
+    def _generate_positive_set(self) -> None:
+        """Build positive dictionary lookup for exact negative sampling."""
+        self.positive_sets = set(zip(self.mat.row, self.mat.col))
```

Comment:
I think this name might be a bit clearer: 

```suggestion
        self.positives_set = set(zip(self.mat.row, self.mat.col))
```